packages/h5wasm/src/h5wasm-api.ts

```diff
@@ -67,10 +66,13 @@ export class H5WasmApi extends DataProviderApi {
       throw new Error('Compression filter not supported');
     }
 
-    // h5wasm returns integers for bool and BigInt for (u)int64
-    // So we use to_array instead to have bool and numbers resp.
-    if (hasBoolType(dataset) || hasInt64Type(dataset)) {
+    /* h5wasm returns bigints for (u)int64 dtypes, so we use `to_array` to get numbers instead.
+     * We do this only for datasets that are supported by at least one visualization (other than `RawVis`),
+     * so for (u)int64 scalars/arrays, and for compound datasets with at least one (u)int64 field (`MatrixVis`). */
+    if (hasInt64Type(dataset)) {
       const rawValue = h5wDataset.to_array();
+
+      // `to_array` returns nested JS arrays for nD datasets, so we need to re-flatten them
```

Comment:
Big downside of going through `to_array`. Would be better if this method had a `flatten` parameter, but I'd still prefer supporting `bigint` in the long run.